### PR TITLE
@idearium/amqp

### PR DESCRIPTION
**Pull request by smebberson**
This PR is the first version of the new @idearium/amqp package which will replace all of the RabbitMQ related code on the legacy branch.

## Related PRs

- [ ] idearium/idearium-lib#349
- [ ] idearium/idearium-lib#347
- [ ] idearium/rest-image#32
- [ ] idearium/ras-rsso#177

## Testing

- [x] Review the code, consider the implementation.
- [x] Review the testing steps and ensure they're thorough enough.
- [x] Review the updated documentation.
- [ ] Complete testing steps on idearium/ras-rsso#177.

## Deployment

- [x] Release @idearium/certs.
- [x] Release @idearium/log.
- [x] Version bump @idearium/certs and @idearium/log.
- [x] Release this package.

### CODE REVIEW

File: packages/amqp/package.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        "@idearium/log": "2.1.0-beta.1",
+        "@idearium/promise-all-settled": "1.0.0",
+        "@idearium/safe-promise": "1.0.0",
+        "amqplib": "0.10.3"
+    },
+    "scripts": {
+        "test": "jest"
+    },
+    "engines": {
+        "node": ">=12"
```
Comment (allanchau) [inline packages/amqp/package.json:19]: I think we should set this to 18 (current LTS) or at the very least 14 which is in maintenance for another month.

**Review by allanchau — CHANGES_REQUESTED** [review]
Minor change to the package.json

File: packages/amqp/package.json (hunk 1 above)
Comment (smebberson) [reply packages/amqp/package.json]: @allanchau, I've updated this to `>=16`.

**Comment by smebberson** [comment]
@allanchau, are you happy to approve this now for release?

**Review by allanchau — APPROVED** [review]

**Pull request merged**